zmon-controller-app/src/main/java/org/zalando/zmon/api/EntityApi.java

```diff
@@ -137,6 +144,16 @@ public int deleteEntity(@PathVariable(value = "id") String id) {
         List<String> deleted = entitySprocs.deleteEntity(id, teams, authService.getUserName());
 
         if (!deleted.isEmpty()) {
+            try {
+                JsonNode e = mapper.readValue(deleted.get(0), JsonNode.class);
+                String type = e.get("type").textValue().toLowerCase();
+                Double created = new Double(e.get("created").doubleValue() * 1000);
+                long duration = (new Date()).getTime() - created.longValue();
+                Timer timer = metricRegistry.timer("controller.entity-lifetime." + type);
+                timer.update(duration, TimeUnit.MILLISECONDS);
+            } catch (IOException ex) {
```

Comment:
I'm not sure, how important this part is, but if it is optional, maybe it would be reasonable to catch all exceptions, to avoid showing error to the user if something went wrong with this MetricRegistry or JSON parsing or whatever.